pkg/operator/platform/platform.go

```diff
@@ -59,3 +67,55 @@ func getClient(explicitKubeconfig string) (client.Client, error) {
 
 	return dynamicClient, nil
 }
+
+func GetFeatureGates() (featuregates.FeatureGate, error) {
+	stop := make(chan struct{})
+	ctx := context.Background()
+	ctx, cancelFn := context.WithCancel(ctx)
+	go func() {
+		defer cancelFn()
+		<-stop
+	}()
+
+	config, err := crtconfig.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kube config: %v", err)
+	}
+	clientSet, err := configclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	configInformers := configinformers.NewSharedInformerFactory(clientSet, 10*time.Minute)
+	desiredVersion := computeClusterOperatorVersions()
+	missingVersion := desiredVersion
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
+	// Notice the nil EventRecorder passed, is this okay?
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		nil,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(stop)
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		log.Info("FeatureGates initialized", "knownFeatures", featureGates.KnownFeatures())
+	case <-time.After(1 * time.Minute):
+		log.Error(nil, "timed out waiting for FeatureGate detection")
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+
+	featureGates, err := featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return nil, err
+	}
+	return featureGates, nil
+}
+
+func computeClusterOperatorVersions() string {
+	currentVersion := os.Getenv("RELEASE_VERSION")
```

Comment:
What if this is not set?

Reply:
A string cannot be nil in golang, it will instead be an empty string. So, in such a case it will be looking for the cluster version to be an empty string. In such a case, I expect a lot of other things will also be broken. The good news is RELEASE_VERSION is defined in the [03-deployment.yaml](https://github.com/openshift/cloud-credential-operator/blob/ee67cc6b1a8cbd2248a2536b0dfa93e18d1bfda2/manifests/03-deployment.yaml#L71) and should always be non-empty.